modules/demo/client-server/routes/cudf.js

```diff
@@ -0,0 +1,277 @@
+    }
+  });
+  datasetLoaded = 'mortgage';
+  let url =
+    'https://raw.githubusercontent.com/rapidsai/cuxfilter/GTC-2018-mortgage-visualization/javascript/demos/GTC%20demo/src/data/zip3-ms-rhs-lessprops.json';
+
+  fetch(url, settings).then(res => res.json()).then((json) => {
+    gjson = json;
+    console.log('readMortgageData complete');
+    var t1 = new Date().getTime();
+    console.log('Time Taken:', t1 - t0, 'ms');
+    callback(true);
+  });
+}
+
+// read Uber Dataset from disk using cudf.DataFrame.readCSV, and fetch geoJSON file
+function readUberData(callback) {
+  var t0 = new Date().getTime();
+  console.log('readUberData');
+  df            = DataFrame.readCSV({
+    header: 0,
+    sourceType: 'files',
+    // sources: [path.join(__dirname, '../public/data/san_fran_uber.csv')],
+    sources: [('modules/demo/client-server/public/data/san_fran_uber.csv')],
+    dataTypes: {
+      sourceid: 'int16',
+      dstid: 'int16',
+      month: 'int8',
+      day: 'int8',
+      start_hour: 'int8',
+      end_hour: 'int8',
+      travel_time: 'float32'
+    }
+  });
+  datasetLoaded = 'uber';
+
+  let url = 'https://movement.uber.com/travel-times/1_censustracts.json'
+  fetch(url, settings).then(res => res.json()).then((json) => {
+    gjson = json;
+    console.log('readUberData complete');
+    var t1 = new Date().getTime();
+    console.log('Time Taken:', t1 - t0, 'ms');
+    callback(true);
+  });
+}
+
+// release data from GPU memory
+function deleteData() {
+  delete df;
+  datasetLoaded = '';
+}
+
+/**
+ * transform Arrow.Table to the format
+ * [index[0]: {property: propertyValue, ....},
+ * index[1]: {property: propertyValue, ....},
+ * ...] for easier conversion to geoJSON object
+ *
+ * @param data Arrow.Table
+ * @param by str, column name
+ * @param params [{}, {}] result of Arrow.Table.toArray()
+ * @returns [index:{props:propsValue}]
+ */
+function transform_data(data, by, params) {
+  return data.reduce((a, v) => {
+    a[v[[by]]] = params.reduce((res, x) => {
+      res[x] = v[x];
+      return res
+    }, {});
+    return a;
+  }, {});
+}
+
+/**
+ * convert an Arrow table to a geoJSON to be consumed by DeckGL GeoJSONLayer. Arrow table results
+ * from cudf.DataFrame.toArrow() function.
+ *
+ * @param data Arrow.Table
+ * @param by str, column name to be matched to the geoJSONProperty in the gjson object
+ * @param properties [] of column names, properties to add from data to result geoJSON
+ * @param geojsonProp str, property name in gjson object to be mapped to `by`
+ * @returns geoJSON object consumable by DeckGL GeoJSONLayer
+ */
+function convertToGeoJSON(data, by, properties, geojsonProp) {
+  data     = transform_data(data.toArray(), by, properties);
+  tempjson = [];
+  gjson.features.forEach((val) => {
+    if (val.properties[geojsonProp] in data) {
+      tempjson.push({
+        type: val.type,
+        geometry: val.geometry,
+        properties: {...val.properties, ...data[val.properties[geojsonProp]]}
```

Comment:
Are the `x in data` and indexing `data[prop]` statements working as expected? I didn't think the field names are JS properties of the Arrow Table?

Reply:
`data` is not an Arrow.Table here, it's the result of transform data (see line 115 above `if`), which is of the format {index:{}, ...}